openpype/hosts/tvpaint/plugins/publish/extract_sequence.py

```diff
@@ -267,38 +341,75 @@ def render_review(
+            george_script_lines.extend(["tv_markin {}".format(origin_mark_in)])
+            george_script_lines.extend(["tv_markout {}".format(origin_mark_out)])
+
+        george_script_lines = "\n".join(george_script_lines)
+        execute_george_through_file(george_script_lines)
+
+        output_filepaths_by_frame_index = {}
+
+        # Check if frames has been properly created on the disk
+        for frame_index in range(mark_in, mark_out + 1):
+            filename = filename_template.format(frame=frame_index)
             filepath = os.path.join(output_dir, filename)
 
-            output_filepaths_by_frame_idx[frame_idx] = filepath
+            output_filepaths_by_frame_index[frame_index] = filepath
 
+            # Ensure the frame has been properly created
             if not os.path.exists(filepath):
+                if export_type == "camera" and frame_index == mark_out:
+                    # There is a bug in TVPaint, it seems to NOT render the last frame in camera mode sometimes
+
+                    # TODO: We should recreate a clean george_script_lines with only info to render the last frame
+                    # Replace the export filepath
+                    last_frame_filename = filename_template.format(frame=mark_out)
+                    george_script_lines = re.sub(origin_first_filename, last_frame_filename, george_script_lines)
+
+                    # set the mark in to mark_out
+                    if "tv_markin" not in george_script_lines:
+                        george_script_lines = "{}\n{}".format("tv_markin {}".format(mark_out), george_script_lines)
+                        george_script_lines = "{}\n{}".format(george_script_lines,
+                                                              "tv_markin {}".format(origin_mark_in))
+                    else:
+                        george_script_lines = re.sub("tv_markin {}".format(mark_in), "tv_markin {}".format(mark_out),
+                                                     george_script_lines)
+
+                    george_script_lines = re.sub(tv_export, tv_export.replace(str(export_lenght), "0"),
+                                                 george_script_lines)
+
+                    execute_george_through_file(george_script_lines)
+
                 raise KnownPublishError(
                     "Output was not rendered. File was not found {}".format(
                         filepath
                     )
                 )
 
-            if first_frame_filepath is None:
-                first_frame_filepath = filepath
-
         thumbnail_filepath = None
-        if first_frame_filepath and os.path.exists(first_frame_filepath):
+        if output_filepaths_by_frame_index:
             thumbnail_filepath = os.path.join(output_dir, "thumbnail.jpg")
-            source_img = Image.open(first_frame_filepath)
+            source_img = Image.open(next(iter(output_filepaths_by_frame_index)))
```

Comment:
Erreur lors du test d'esport en playblast, type Scene:
  File "C:\Users\hfarre\quad\OpenPype\openpype\hosts\tvpaint\plugins\publish\extract_sequence.py", line 401, in render_review
    source_img = Image.open(next(iter(output_filepaths_by_frame_index)))
  File "C:\Users\hfarre\quad\OpenPype\.venv\lib\site-packages\PIL\Image.py", line 3242, in open
    fp = io.BytesIO(fp.read())
AttributeError: 'int' object has no attribute 'read'

Reply:
Fixed